contrib/mesos/pkg/executor/executor.go

```diff
@@ -565,6 +542,9 @@ func (k *KubernetesExecutor) launchTask(driver bindings.ExecutorDriver, taskId s
 	k.pods[podFullName] = pod
 
 	// send the new pod to the kubelet which will spin it up
+	if k.isDone() {
```

Comment:
instead of scattering calls to `k.isDone()` throughout the code, could we write something like this:
```
// sendPodUpdate assumes that caller is holding state lock; returns true when update is sent otherwise false
func sendPodUpdate(u *kubelet.PodUpdate) bool {
  if k.isDone() {
    return false
  }
  k.updateChan <- *u
  return true
}
```